util/Random.cpp

```diff
@@ -1,56 +1,62 @@
 #include "Random.h"
 
 #include <boost/date_time/posix_time/posix_time.hpp>
-#include <boost/thread/mutex.hpp>
+#include <mutex>
 
-namespace {
-    GeneratorType gen; // the one random number generator driving the distributions below
 
-    static boost::mutex s_prng_mutex;
+typedef std::lock_guard<std::mutex> lock_guard;
+
+namespace {
+    GeneratorType gen{2462343}; // the one random number generator driving the distributions below. arbitrarily chosen default seed
+    static std::mutex s_prng_mutex;
 }
 
 void Seed(unsigned int seed) {
-    boost::mutex::scoped_lock lock(s_prng_mutex);
-    gen.seed(static_cast<boost::mt19937::result_type>(seed));
+    lock_guard lock(s_prng_mutex);
+    gen.seed(static_cast<GeneratorType::result_type>(seed));
 }
 
 void ClockSeed() {
-    boost::mutex::scoped_lock lock(s_prng_mutex);
+    lock_guard lock(s_prng_mutex);
     boost::posix_time::time_duration diff = boost::posix_time::microsec_clock::local_time().time_of_day();
-    gen.seed(static_cast<boost::mt19937::result_type>(diff.total_milliseconds()));
+    gen.seed(static_cast<GeneratorType::result_type>(diff.total_milliseconds()));
 }
 
-SmallIntDistType SmallIntDist(int min, int max) {
-    boost::mutex::scoped_lock lock(s_prng_mutex);
-    return SmallIntDistType(gen, boost::uniform_smallint<>(min, max));
+int RandInt(int min, int max) {
+    if (min <= max)
+        return min;
+    std::uniform_int_distribution<> dis(min, max);
```

Comment:
C++11 does not guarrantee reproducible distributions accross implementations:

http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2059r0.pdf